lib/ecto_neo4j.ex

```diff
@@ -80,18 +83,15 @@ defmodule Ecto.Neo4j do
     {{_,_,[_head|columns]},type,_} = expr
     {hd(columns), type[:ecto_type]}
   end
-
   def extract_column_with_type(expr), do: expr
 
-  def extract_column(expr), do: expr
-
-
   def build_cypher(query) do
     {type, query_obj} = query
     case type do
       :all ->
-        columns_string = columns(query) |> Enum.uniq |> Enum.map(fn column -> "m.#{column} as #{column}" end)
-        {from, _} = query_obj.from
+        formatter      = fn column -> "m.#{column} as #{column}" end
+        columns_string = query |> columns |> Enum.uniq |> Enum.map(formatter)
+        {from, _}      = query_obj.from
```

Comment:
I like this better than my code, but it seems a bit of a code smell that we're doing all this stuff inside one function (and it was before, too; this refactor just brought it to light).